sdk-docs/project-structure.md

```diff
@@ -0,0 +1,93 @@
+            - `/generated`: This folder is automatically generated when you follow our graphql guide.
+                - `graphql-api.tsx`: It houses auto-generated hooks (functions) that call the graphql api for you.
+            - `/queries`: You should create this folder manually and put all your custom graphql queries in it. The hooks above are generated based on the queries you store in this folder
+                - `[query_name].graphql`: This is one of your custom graphql queries.
+            - `app.spec.tsx`: Unit tests for the App component.
+            - `app.tsx`: The entry-point of your custom code logic. Start coding here!
+            - `index.tsx`: The entry-point of your extension. It will initialize the App component above and you shouldn't do any changes here. Your point of interest is app.tsx.
+
+
+
+
+## Configuration Files
+### iris-app-manifest.ts
+Possibly the most important configuration file that you need to understand.
+It configures the following crucial settings for your app:
+-  `specVersion`: The version of the app manifest file. Should not be changed manually. 
+- `moduleFederationName`: Name of your app including workspace name, automatically parsed from your app-specific package.json.
+- `dependencies`: A full list of runtime dependencies for the app. Automatically parsed from the package.json in root. The list will be security scanned.
+- `devDependencies`: Same as above but for dev-dependencies. These are also security scanned.
+- `validDomains`: A list of any external domains that the app needs to communicate with. The list is audited in the approval step of the app and every external API call must be listed here to get through the verification process.
+- `validForAccountIds`: Put `ALL_ACCOUNTS` in if your app is available to all Trackunit users. If you want to only make it available to certain users, provide a list of account ID's as strings.
+- `activationMode`: Controls how the app can/will be installed.
+    - Enabled means it's enabled on the subscription when this Iris app is approved.
+    - Visible means it will be visible to the customer to choose for them selves to enable/disable it.
+    - FORCE_ENABLE means it cannot be removed by the endcustomer, and will be installed based on the selected subscription package.
+    - DEFAULT_ENABLE means it will be enabled when a customer activates a new subscription package.
+    - ALWAYS_VISIBLE means it will always be visible for all customers.
+    - CONTROLLED_MANUAL_INSTALLATION means it will be controlled by the team that activates the subscription package for the customer, not activating the app.
+    - CONTROLLED_AUTOMATIC_INSTALLATION means it will be controlled by the team that activates the subscription package for the customer, also adding.
+    - All modes will take into consideration both **validForAccountIds** and **marketplace.allowForPackage**
```

Comment:
Do we have a separate page for this with a table? 🤔
I think I remember seeing this somewhere.
If so, let's link to that to avoid creating duplicated things to maintain.

Reply:
Looked through all pages at https://developers.trackunit.com/docs/overview and didn't find it. It's taken from the in-code comments and then elaborated/improved on, so maybe that's why you recognize it. In an optimal scenario I'd ingest it directly from the code comments, but we don't have the tooling for that  right now.

So this is the best we can do for now. Luckily these things don't change too often  :)